src/test/java/com/wolox/training/repositories/BookRepositoryTest.java

```diff
@@ -0,0 +1,91 @@
+package com.wolox.training.repositories;
+
+import com.wolox.training.exceptions.NotFoundException;
+import com.wolox.training.models.Book;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase.Replace;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.boot.test.autoconfigure.orm.jpa.TestEntityManager;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import javax.validation.ConstraintViolationException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@RunWith(SpringRunner.class)
+@DataJpaTest
+@AutoConfigureTestDatabase(replace = Replace.NONE)
+public class BookRepositoryTest {
```

Comment:
Each time you make `repository.save(book)` you are saving a new book into de DB. To make these tests, you should use a [TestEntityManager](https://www.baeldung.com/spring-boot-testing#integration-testing-with-datajpatest) in order to mock the situation in which you have all the books you want in the DB, but without really adding them.